### ISSUE TITLE
[Bugfix 2.0 + Re-Arm] cant control 2nd extruder fan via Gcode

### ISSUE THREAD

**Issue by boelle** [MarlinFirmware/Marlin#12831]
so i have added a 2nd fan in the pins file like this:

#if ENABLED(IS_RAMPS_EFB) || ENABLED(IS_RAMPS_EFF)   // Hotend, Fan, Bed or Hotend, Fan, Fan
    #define FAN_PIN        RAMPS_D9_PIN
	  #define FAN1_PIN       P1_20   // (11)

i can control it from the LCD but when i send Gcode M106 P1 to turn it full on nothing happens

it did thou on my copy from 5/11-2018

what could be wrong this time?

config and pins file attached from todays copy of marlin

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/2729952/Configuration.zip)

**Comment by boelle** [comment]
and just if a compare is needed here are the same files from the 5/11/2018 copy
[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/2729960/Configuration.zip)

**Comment by GMagician** [comment]
Maybe something related to #12820?

**Comment by boelle** [comment]
Not sure, i think @thinkyhead is the better judge of that

but it demonstrates that its always a good idea to keep the last 2 working copies :-D

**Comment by boelle** [comment]
can this be considered a bug? if so i would reword the title by adding [Bug] to it, if its not a bug i will close the issue so i dont take up more space than i need

**Comment by GMagician** [comment]
Have you tryed version before #12820? (change is easy to revert).
Since that function can change used fan may give you an issue...
You may check in your code what's FAN_COUNT value. with 2 fans it should be 2

**Comment by GMagician** [comment]
Sure you put define in correct if branch?
your board is BOARD_RAMPS_14_RE_ARM_EFB

> #if ENABLED(IS_RAMPS_EFB) || ENABLED(IS_RAMPS_EFF)

**Comment by boelle** [comment]
yes i'm sure... look in the zip file, pins file is included

**Comment by GMagician** [comment]
Have you checked what's the FAN_COUNT value?

SERVO0_PIN is same. Maybe fan value is overwritten? Try to change SERVO0_PIN

**Comment by InsanityAutomation** [comment]
Does it work with S255 in the command? I'm not 100% sure full on without setting a speed was ever intended behavior.

**Comment by boelle** [comment]
all i know its works with the copy i downloaded 5/11-2018

so if using the copy from 5/11 i can send: M106 P1 or M106 P1 S255 for full on
M107 P1 turns off

all i had to do to make it work on the 5/11 copy is adding one line here:

#if ENABLED(IS_RAMPS_EFB) || ENABLED(IS_RAMPS_EFF) // Hotend, Fan, Bed or Hotend, Fan, Fan
#define FAN_PIN RAMPS_D9_PIN
#define FAN1_PIN P1_20 // (11)

last line is the one i added

**Comment by DEPREDADOR1888** [comment]
a si es compañero yo estoy usando la la misma versión del marlin y con la ender 4 en la mks sbase y con tu comentario logre hacer funcionar con mi pantalla tft el ventilador al igual que mandando el comando m106   lo cambie en el archivo pins_MKS_SBASE.H y lo que hice fue esto 
--------------------------------------------------------------------
// Heaters / Fans
//
#define HEATER_BED_PIN     P2_05
#define HEATER_0_PIN       P2_07
#define HEATER_1_PIN       P2_06
/*#ifndef RAMPS_D9_PIN
  #define RAMPS_D9_PIN          P2_04
#endif*/
#define FAN_PIN P2_04
---------------------------------------------------------------------
y ya con eso logre hacer funcionar los ventiladores gracias tenia rato con ese problema y ya quedo resuelto con esto saludos.

**Comment by boelle** [comment]
i tried to use google translate to translate from spanish but it made no sense

**Comment by boelle** [comment]
think i got it...

you have the mks sbase and can control the extra fan from the lcd, so can i

but i can not control it via gcode

only thing i cant see in your pin example is where your extra fan is defined, that is what i did, i defined an extra part cooling fan that can be controlled seperately from the standard one

**Comment by boelle** [comment]
i want @thinkyhead to answer this one:

Is it not possible to add the extra part cooling fan to the RAMPS_EFB setup like i did and still be able to control it? or do i need to add something extra?
